### PR TITLE
Parse typed Mapfile object

### PR DESCRIPTION
**Pull request by b4l**
GSSDI-15
Don't know if this is the desired and the right way to do it, so i am looking for feedback!

### CODE REVIEW

File: src/MapfileStyleParser.ts [hunk 1]
```diff
@@ -526,44 +530,44 @@ export class MapfileStyleParser implements StyleParser {
...
   /**
-   * Get the GeoStyler-Style Basic Symbolizer Parameter from an Mapfile STYLE.
+   * Get the GeoStyler-Style Basic Symbolizer Parameter from an Mapfile STYLE / LABEL.
    *
-   * @param {object} styleParameters The Mapfile Style
+   * @param {MapfileStyle | MapfileLabel} styleParameters The Mapfile Style
    * @return {Symbolizer} The GeoStyler-Style Symbolizer Parameters
    */
-  getBaseSymbolizerFromStyle(styleParameters: any): any {
+  getBaseSymbolizerFromMapfileStyle(styleParameters: any): Symbolizer {
```
Comment (ger-benjamin) [inline src/MapfileStyleParser.ts:570]:
```suggestion
  getBaseSymbolizerFromMapfileStyle(styleParameters: MapfileStyle | MapfileLabel): Symbolizer {
```

File: src/MapfileStyleParser.ts [hunk 2]
```diff
@@ -581,30 +585,30 @@ export class MapfileStyleParser implements StyleParser {
...
+    mapfileClass: MapfileClass,
     mapfileLayerType: string,
     mapfileLayerLabelItem: string
   ): Symbolizer[] {
     const symbolizers = [] as Symbolizer[];
     // Mapfile STYLE
-    if (mapfileClass.style) {
-      mapfileClass.style.forEach((styleParameters: any) => {
+    if (mapfileClass.styles) {
+      mapfileClass.styles.forEach((mapfileStyle: MapfileStyle) => {
```
Comment (ger-benjamin) [inline src/MapfileStyleParser.ts:601]:
```suggestion
      mapfileClass.styles.forEach((mapfileStyle) => {
```
I guess it should work without the type (implicite type)

File: src/mapfile2js/parse.ts [hunk 3]
```diff
@@ -6,84 +6,224 @@ import { checkBlockEndSum } from './parse/checkBlockEndSum';
...
-  let result = {};
-  const lineObjects: Array<any> = [];
+export interface Mapfile {
+  map: MapfileMap;
+}
 
-  // stack to keep track of blocks
-  const blocks = [result];
 
+interface MapfileSymbolset {
```
Comment (ger-benjamin) [inline src/mapfile2js/parse.ts:28]: Can you move all the mapfile element definition in an external `mapfileTypes.ts` file ?

File: src/MapfileStyleParser.ts (hunk 2 above)
Comment (b4l) [reply src/MapfileStyleParser.ts:601]: That is right. I removed the explicit typing.

File: src/MapfileStyleParser.ts (hunk 1 above)
Comment (b4l) [reply src/MapfileStyleParser.ts:570]: Done

File: src/mapfile2js/parse.ts (hunk 3 above)
Comment (b4l) [reply src/mapfile2js/parse.ts:28]: Done

**Comment by marcjansen** [comment]
Nice!

**Pull request merged**